<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordGeneratorDialog</class>
 <widget class="QDialog" name="PasswordGeneratorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>350</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Password Generator</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLineEdit" name="passwordPreview">
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="styleSheet">
      <string>QLineEdit {
    font-family: 'Courier New';
    font-size: 16px;
    font-weight: bold;
    padding: 10px;
}</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Length:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="lengthLayout">
       <item>
        <widget class="QSlider" name="lengthSlider">
         <property name="minimum">
          <number>8</number>
         </property>
         <property name="maximum">
          <number>64</number>
         </property>
         <property name="value">
          <number>16</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="lengthSpinBox">
         <property name="minimum">
          <number>8</number>
         </property>
         <property name="maximum">
          <number>64</number>
         </property>
         <property name="value">
          <number>16</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Character Sets</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QCheckBox" name="upperCheck">
        <property name="text">
         <string>Uppercase (A-Z)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="lowerCheck">
        <property name="text">
         <string>Lowercase (a-z)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="numbersCheck">
        <property name="text">
         <string>Numbers (0-9)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="symbolsCheck">
        <property name="text">
         <string>Symbols (!@#$%...)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="generateButton">
     <property name="text">
      <string>Regenerate</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <enum>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PasswordGeneratorDialog</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PasswordGeneratorDialog</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui>
